code/datums/uplink/uplink_categories.dm

```diff
@@ -1,49 +1,117 @@
 /datum/uplink_category
 	var/name = ""
 	var/list/datum/uplink_item/items
+	var/list/datum/antagonist/antag_roles  // added this so category wide restrictions can be placed to make it eaiser to set restrictions
 
 /datum/uplink_category/New()
 	..()
 	items = list()
+	antag_roles = list()
```

Comment:
Well, my concern was you can see blank categories. This is a special case where an antag type is introduced that can only see one category. I tried to ensure that having categories be given the New() proc was a last resort. I tried placing the antag roles under the prototype and the items within and it didn't work. If you are okay with players being able to see blank categories, then I can remove it and instead assign it to the prototype item, but personally I don't like being able to see categories that I can't buy from.